src/core/include/ngraph/opsets/opset.hpp

```diff
@@ -47,18 +47,19 @@ class NGRAPH_API OpSet : public ov::OpSet {
         ov::OpSet::insert<OP_TYPE>(OP_TYPE::get_type_info_static().name);
     }
 
-    ngraph::FactoryRegistry<ngraph::Node>& get_factory_registry() {
-        return m_factory_registry;
-    }
+    ngraph::FactoryRegistry<ngraph::Node> get_factory_registry();
+
+private:
+    ngraph::FactoryRegistry<ngraph::Node> m_factory_registry;
 };
 
-const NGRAPH_API OpSet& get_opset1();
-const NGRAPH_API OpSet& get_opset2();
-const NGRAPH_API OpSet& get_opset3();
-const NGRAPH_API OpSet& get_opset4();
-const NGRAPH_API OpSet& get_opset5();
-const NGRAPH_API OpSet& get_opset6();
-const NGRAPH_API OpSet& get_opset7();
-const NGRAPH_API OpSet& get_opset8();
-const NGRAPH_API OpSet& get_opset9();
+const NGRAPH_API OpSet& get_opset1(const std::string& name = "default");
+const NGRAPH_API OpSet& get_opset2(const std::string& name = "default");
+const NGRAPH_API OpSet& get_opset3(const std::string& name = "default");
+const NGRAPH_API OpSet& get_opset4(const std::string& name = "default");
+const NGRAPH_API OpSet& get_opset5(const std::string& name = "default");
+const NGRAPH_API OpSet& get_opset6(const std::string& name = "default");
+const NGRAPH_API OpSet& get_opset7(const std::string& name = "default");
+const NGRAPH_API OpSet& get_opset8(const std::string& name = "default");
+const NGRAPH_API OpSet& get_opset9(const std::string& name = "default");
```

Comment:
why do we name `name` here?

Reply:
It was done in order to pass name to CC factory to collect statistics. Each factory should have unique name. In case of names intersection CC won't eliminate all possible code, but it's also expected in order to guarantee backward compatibility (and CC factories will have "default" names)

For ngraph::OpSet name was introduced to enable CC for our internal code, default value is required for backward compatibility

Reply:
@ilyachur pointed me that `get_opset#` return const reference to static object, so that there is no need to provide name outside the function. In this case OpSet will have constant name, and `get_opset#` functions won't have an argument